app/controllers/assignments_controller.rb

```diff
@@ -0,0 +1,95 @@
+class AssignmentsController < ApplicationController
+    def index
+        @user = User.find(params[:user_id])
+        ensure_proper_user(@user)
+        @course = @user.courses.find(params[:course_id])
+        @assignments = @course.assignments
+    end
+
+    def new
+        @user = User.find(params[:user_id])
+        ensure_proper_user(@user)
+        @course = @user.courses.find(params[:course_id])
+        @assignment = @course.assignments.new
+    end
+
+    def create
+        @user = User.find(params[:user_id])
+        ensure_proper_user(@user)
+        @course = @user.courses.find(params[:course_id])
+        @assignment = @course.assignments.create(assignment_params)
+        if @assignment.save
+            redirect_to user_course_assignment_path(@user, @course, @assignment)
+        else
+            render 'new'
+      end
+    end
+
+    def show
+        @user = User.find(params[:user_id])
+        ensure_proper_user(@user)
+        @course = @user.courses.find(params[:course_id])
+        @assignment = @course.assignments.find(params[:id])
+        @submissionTag = helpers.hashID(@assignment.id)
+    end
+
+    def destroy
+        @user = User.find(params[:user_id])
+        ensure_proper_user(@user)
+        @course = @user.courses.find(params[:course_id])
+        @assignment = @course.assignments.find(params[:id])
+        @assignment.destroy
+        redirect_to user_course_path(@user, @course)
```

Comment:
Should this maybe take the user to a listing of the assignments instead of to the course?